public/app/features/transformers/extractFields/extractFields.ts

```diff
@@ -79,7 +79,7 @@ export function addExtractedFields(frame: DataFrame, options: ExtractFieldsOptio
       if (filteredPaths.length > 0) {
         filteredPaths.forEach((path: JSONPath) => {
           const key = path.alias && path.alias.length > 0 ? path.alias : path.path;
-          newObj[key] = get(obj, path.path) ?? 'Not Found';
+          newObj[key] = get(obj, path.path) ?? (isNumber(obj) || isBoolean(obj) ? obj : 'Not Found');
```

Comment:
Note: isString is handled prior to this https://github.com/grafana/grafana/blob/main/public/app/features/transformers/extractFields/extractFields.ts#L62